src/main/java/com/bobocode/petros/container/ApplicationAnnotationContainer.java

```diff
@@ -20,20 +19,16 @@ public ApplicationAnnotationContainer(String packageName) {
     @Override
     public <T> T getDependency(String name, Class<T> clazz) {
         DependencyDefinition definition = keyByDependencyDefinitionName(name);
-        return (T) dependencyMap.values().stream()
-                .filter(obj -> obj.getClass().getName().toLowerCase().equals(name.toLowerCase())
-                        && obj.getClass().equals(clazz))
-                .findFirst()
-                .orElseThrow(NoSuchElementException::new);
+        return (T) dependencyMap.get(definition);
     }
 
 
     @Override
     public <T> T getDependency(Class<T> clazz) {
-        if (isUniqueDependency(clazz)){
-            return getDependencyFromMap(clazz);
-        } else {
+        if (isNonUniqueDependency(clazz)){
             throw new NoUniqueDependecyException(clazz.getName());
+        } else {
+            return getDependencyFromMap(clazz);
```

Comment:
I think we can return just Object without Optional.

Reply:
fixed